src/GameMain/BoardUtility.java

```diff
@@ -73,6 +73,62 @@ public static void loadBoardUtility(ArrayList<SpriteClasses.TankAI> enemy1,
         tank = tank1;
     }
 
+    public static void updatePowerUps() {
+        Iterator<PowerUp> iterator = powerUps.iterator();
+
+        while (iterator.hasNext()) {
+            PowerUp p = iterator.next();
+            p.updateAnimation();
+
+            BlockType type = BlockType.getTypeFromInt(p.getType());
+            Rectangle r1 = tank.getBounds();
+            Rectangle r2 = p.getBounds();
+
+            if (System.currentTimeMillis() - p.getLoadTime() > 10000) {
+                iterator.remove();
+            }
+
+            if (r1.intersects(r2)) {
+                iterator.remove();
+                SoundUtility.powerupPick();
```

Comment:
Consider passing a sound player as a parameter or using a callback for better decoupling